nautobot/core/celery/backends.py

```diff
@@ -54,14 +55,21 @@ def _get_extended_properties(self, request, traceback):
             if traceback is not None:
                 traceback = sanitize(traceback)
 
+            # Preserve the JobResult data behavior from Nautobot 2.0 through 2.2 (wherein the Job itself was the task)
+            # by manipulating `task_name` and `task_args` to hide the fact that we are now calling
+            # `run_job.apply(args=[JobClass.class_path, ...])` instead of `JobClass.apply(args=[...])`.
+            if task_name == "nautobot.extras.jobs.run_job" and task_args:
+                task_name = task_args[0]
+                task_args = task_args[1:]
```

Comment:
```suggestion
                task_name = task_args.pop(0)
```

Reply:
This won't work if `task_args` is a tuple. Would that ever be the case?

Reply:
I think it's always a list (since we're passing things around in JSON) but good question.

Reply:
I think I'll leave this alone just to be safe, as I can't tell from a quick read of the Celery code whether there's any particular type assertion here.